source/common/network/io_socket_error_impl.cc

```diff
@@ -25,6 +25,8 @@ Api::IoError::IoErrorCode IoSocketError::getErrorCode() const {
     return IoErrorCode::Interrupt;
   case EADDRNOTAVAIL:
     return IoErrorCode::AddressNotAvailable;
+  case EBADF:
```

Comment:
If we can land this as a quic-only change I'll be way happier.  If we opt to get rid of isQuic let's just split this out if it's the only other non-quic code in the PR

Reply:
Q: Why do we need this again? I remember this in a previous PR and we removed it. Shouldn't this never happen / be an assert somewhere?

Reply:
Good point! This was added because a test didn't mock out sys calls and previously there was an ASSERT in the `default:` block. Now the test shouldn't crash since we down-graded the severity.